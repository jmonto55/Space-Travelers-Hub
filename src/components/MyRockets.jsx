import { useSelector } from 'react-redux';

const MyRockets = () => {
  const { rocketsList } = useSelector((store) => store.rockets);
  const reservedRockets = rocketsList.filter((rokt) => rokt.reserved === true);

  return (
    <div className="w-1/2">
      <h2 className="font-semibold text-2xl mb-3">My Rockets</h2>
      <ul className="border-solid border-t-2 border-neutral-300">
        {reservedRockets.map((rokt) => <li className="border-solid border-2 border-t-0 border-neutral-300 p-5 text-lg" key={rokt.id}>{rokt.name}</li>)}
      </ul>
    </div>
  );
};

export default MyRockets;
