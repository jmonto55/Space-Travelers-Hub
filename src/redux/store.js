import { configureStore, combineReducers } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionReducer from './missions/missionsSlice';

const rootReducer = combineReducers({ rockets: rocketsReducer, missions: missionReducer });

const store = configureStore({ reducer: rootReducer });

export default store;
