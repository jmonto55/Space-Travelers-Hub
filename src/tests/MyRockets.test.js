import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import MyRockets from '../components/MyRockets';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the MyRockets component', () => {
  it('Should render a Snapshot of the MyRockets component', () => {
    const myrockets = render(
      <Provider store={store}>
        <MyRockets />
      </Provider>,
    );
    expect(myrockets).toMatchSnapshot();
  });

  test('Should test if the text "My Rockets" is inside the MyRockets component', () => {
    render(
      <Provider store={store}>
        <MyRockets />
      </Provider>,
    );
    expect(screen.getByText('My Rockets')).toBeInTheDocument();
  });
});
