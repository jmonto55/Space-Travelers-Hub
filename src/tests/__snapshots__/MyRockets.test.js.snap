// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the MyRockets component Should render a Snapshot of the MyRockets component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-1/2"
      >
        <h2
          class="font-semibold text-2xl mb-3"
        >
          My Rockets
        </h2>
        <ul
          class="border-solid border-t-2 border-neutral-300"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-1/2"
    >
      <h2
        class="font-semibold text-2xl mb-3"
      >
        My Rockets
      </h2>
      <ul
        class="border-solid border-t-2 border-neutral-300"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
